assets/app/view/game/buy_trains.rb

```diff
@@ -300,7 +300,7 @@ def render
           children << h(:div, issue_str)
         end
 
-        if @step.can_finance?(@corporation)
+        if @step.respond_to?(:can_finance?) && @step.can_finance?(@corporation)
```

Comment:
I purposely avoided doing needing to do this by adding the method to the buy_train step. Looking at 1877's buy_sell_par_shares step, it copies over all the methods from the buy_train step. I think it should copy over this new method too.

Reply:
That would work too. I'd done the `respond_to?` check as we've already got a zillion tests like this in the UI code.

I've had a quick grep and can't see any other games that have a BuyTrain action in a step that's not inheriting from Step::BuyTrain.

Reply:
Yeah, I'd prefer to follow the pattern of what 1877 already did, rather than in this one case use `respond_to?`. I don't see a reason to make this one a special case that behaves differently. Do you want to make that change or should I?

Reply:
I probably won't have time to do this for a couple of days, so go for it if you can do it sooner.